transport/socks5/stream_dialer.go

```diff
@@ -23,23 +23,29 @@ import (
 	"github.com/Jigsaw-Code/outline-sdk/transport"
 )
 
+type Credentials struct {
+	Username string
+	Password string
+}
+
 // NewStreamDialer creates a [transport.StreamDialer] that routes connections to a SOCKS5
 // proxy listening at the given [transport.StreamEndpoint].
-func NewStreamDialer(endpoint transport.StreamEndpoint) (transport.StreamDialer, error) {
+func NewStreamDialer(endpoint transport.StreamEndpoint, cred Credentials) (transport.StreamDialer, error) {
```

Comment:
Credentials should be a pointer, so we can pass nil.

You must check the length of the credentials, which must be <=255